chichiwang/exercise-js
# Move Layout rendering to the data router

## Description
Linked to Issue: #81
Move the responsibility of rendering the `Layout` from the pages themselves to the `dataRouter`. The Layout is refactored to render a [React Router Outlet](https://reactrouter.com/en/main/components/outlet) rather than accepting children. The purpose of this is that the Nav Bar accent animation will persist across client-side routing, but it comes with other benefits:

* Pages are no longer responsible for determining which layout they should render with
  * This means pages become more composable from a higher level
  * This also means snapshots will not need to be updated for every page that uses a given layout whenever that layout is updated

## Changes
* Change the order of imports in `app/App/index.tsx` so that the styles are imported after the type definition for `Manifest`
* Re-structure `app/dataRoutes/index.ts` so that all current page routes are nested under a layout route definition
* Refactor pages so that they are not explicitly wrapped in the `<Layout>` component
  * `pages/Counter/index.tsx`
  * `pages/ErrorPage/index.tsx`
  * `pages/Home/index.tsx`
  * `pages/NotFound/index.tsx`
* Refactor `pages/Layout/index.tsx`
  * No longer accepts a prop `children`
  * Renders a [React Router Outlet](https://reactrouter.com/en/main/start/concepts#outlets) component instead

## Steps to QA
* Pull down and switch to this branch
* Verify the app runs, and that all routes still look correct
* Navigate using links in browser, verify the nav bar accent maintains state across page navigation (including the back and forward browser buttons)
* Navigate to a non-existent route to verify the `NotFound` page renders with the layout
* Introduce an error on a page (ex: `/counter`) and navigate to that page in the browser to verify the 500 page renders with the layout
